src/zarr/store/core.py

```diff
@@ -54,13 +56,11 @@ def __str__(self) -> str:
     def __repr__(self) -> str:
         return f"StorePath({self.store.__class__.__name__}, {str(self)!r})"
 
-    def __eq__(self, other: Any) -> bool:
+    def __eq__(self, other: object) -> bool:
         try:
-            if self.store == other.store and self.path == other.path:
-                return True
+            return bool(self.store == other.store and self.path == other.path)  # type: ignore[attr-defined]
         except Exception:
```

Comment:
Perhaps catch only `AttributeError` instead of `Exception`?

Reply:
seems like a good improvement

Reply:
Applied `Exception` → `(AttributeError, TypeError)`